public/app/features/correlations/Forms/AddCorrelationForm.tsx

```diff
@@ -0,0 +1,123 @@
+import { css } from '@emotion/css';
+import React, { useCallback } from 'react';
+import { Controller } from 'react-hook-form';
+
+import { DataSourceInstanceSettings, GrafanaTheme2 } from '@grafana/data';
+import { DataSourcePicker } from '@grafana/runtime';
+import { Button, Field, HorizontalGroup, PanelContainer, useStyles2 } from '@grafana/ui';
+import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+
+import { useCorrelations } from '../useCorrelations';
+
+import { CorrelationDetailsFormPart } from './CorrelationDetailsFormPart';
+import { FormDTO } from './types';
+import { useCorrelationForm } from './useCorrelationForm';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  panelContainer: css`
+    position: relative;
+    padding: ${theme.spacing(1)};
+    margin-bottom: ${theme.spacing(2)};
+  `,
+  linksToContainer: css`
+    flex-grow: 1;
+    /* This is the width of the textarea minus the sum of the label&description fields,
+     * so that this element takes exactly the remaining space and the inputs will be
+     * nicely aligned with the textarea
+    **/
+    max-width: ${theme.spacing(80 - 64)};
```

Comment:
We may need something more robust or different layouting as it may not work well with i18n but we can revise it later.

Reply:
yeah, but tbh the whole form layout is temporary as we don't have anything else there. the whole thing will change once we add all the correlations settings in the form.